content/src/storage/ContentStorageFactory.ts

```diff
@@ -2,10 +2,34 @@ import path from 'path'
 import { Environment, EnvironmentConfig } from '../Environment'
 import { ContentStorage } from './ContentStorage'
 import { FileSystemContentStorage } from './FileSystemContentStorage'
+import { S3ContentStorage } from './S3ContentStorage'
 
 export class ContentStorageFactory {
+  static create(env: Environment): Promise<ContentStorage> {
+    if (this.shouldStoreInS3(env)) {
+      return this.s3(env)
+    } else {
+      return this.local(env)
+    }
+  }
+
+  static shouldStoreInS3(env: Environment): boolean {
+    return (
+      !!env.getConfig(EnvironmentConfig.S3_STORAGE_ACCESS_KEY_ID) &&
+      !!env.getConfig(EnvironmentConfig.S3_STORAGE_SECRET_ACCESS_KEY) &&
+      !!env.getConfig(EnvironmentConfig.S3_STORAGE_BUCKET)
+    )
```

Comment:
To enable S3 we should only validate if a `EnvironmentConfig.S3_STORAGE_BUCKET` is present. The good practice in deployments is to use [instanceProfile](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2.html) which do not require any access keys, it is like "the machine's role".. It is better in terms of security because it uses the origin (instance ip) of the requests authenticate against the role.